warp10/src/main/java/io/warp10/script/ScriptRunner.java

```diff
@@ -39,265 +68,232 @@
-    Configuration.RUNNER_KAFKA_GROUPID,
-    Configuration.RUNNER_KAFKA_COMMITPERIOD,
-    Configuration.RUNNER_KAFKA_NTHREADS,
-    Configuration.RUNNER_NTHREADS,
-    Configuration.RUNNER_ENDPOINT,    
-    Configuration.RUNNER_ID,
+      Configuration.RUNNER_KAFKA_ZKCONNECT,
+      Configuration.RUNNER_KAFKA_TOPIC,
+      Configuration.RUNNER_KAFKA_GROUPID,
+      Configuration.RUNNER_KAFKA_COMMITPERIOD,
+      Configuration.RUNNER_KAFKA_NTHREADS,
+      Configuration.RUNNER_NTHREADS,
+      Configuration.RUNNER_ENDPOINT,
+      Configuration.RUNNER_ID,
   };
 
   /**
    * Required properties for the distributed 'scheduler' role of ScriptRunner
    */
   public static final String[] REQUIRED_PROPERTIES_SCHEDULER = {
-    Configuration.RUNNER_KAFKA_BROKERLIST,
-    Configuration.RUNNER_KAFKA_TOPIC,
-    Configuration.RUNNER_KAFKA_POOLSIZE,
-    Configuration.RUNNER_ROOT,
-    Configuration.RUNNER_SCANPERIOD,
-    Configuration.RUNNER_MINPERIOD,
-    Configuration.RUNNER_ID,
-    Configuration.RUNNER_ZK_QUORUM,
-    Configuration.RUNNER_ZK_ZNODE,
+      Configuration.RUNNER_KAFKA_BROKERLIST,
+      Configuration.RUNNER_KAFKA_TOPIC,
+      Configuration.RUNNER_KAFKA_POOLSIZE,
+      Configuration.RUNNER_ROOT,
+      Configuration.RUNNER_SCANPERIOD,
+      Configuration.RUNNER_MINPERIOD,
+      Configuration.RUNNER_ID,
+      Configuration.RUNNER_ZK_QUORUM,
+      Configuration.RUNNER_ZK_ZNODE,
   };
 
   private final boolean isScheduler;
-  private final boolean isWorker;
   private final boolean isStandalone;
   private final KeyStore keystore;
-  
+
   private final byte[] runnerPSK;
-  
+
   private final boolean runAtStartup;
-  
+
   public ScriptRunner(KeyStore keystore, Properties config) throws IOException {
-      
+
     //
     // Extract our roles
     //
-    
+
     Preconditions.checkNotNull(config.getProperty(Configuration.RUNNER_ROLES), "Property '" + Configuration.RUNNER_ROLES + "' MUST be set.");
-    
+
     String[] roles = config.getProperty(Configuration.RUNNER_ROLES).split(",");
-    
+
     if (roles.length > 2) {
       throw new IOException("Role can only be 'standalone' or either or both 'scheduler' and 'worker'.");
     }
 
     Set<String> configuredRoles = new HashSet<String>();
-    
-    for (String role: roles) {
-      configuredRoles.add(role);
-    }
+    configuredRoles.addAll(Arrays.asList(roles));
 
-    if (configuredRoles.contains("standalone")) {
-      isStandalone = true;
-    } else {
-      isStandalone = false;
-    }
-    
-    if (configuredRoles.contains("scheduler")) {
-      isScheduler = true;
-    } else {
-      isScheduler = false;
-    }
-    
-    if (configuredRoles.contains("worker")) {
-      isWorker = true;
-    } else {
-      isWorker = false;
-    }
-  
-    if (isStandalone || isScheduler) {      
+    isStandalone = configuredRoles.contains("standalone");
+
+    isScheduler = configuredRoles.contains("scheduler");
+
+    boolean isWorker = configuredRoles.contains("worker");
```

Comment:
Why redefining isWorker here?

Reply:
isWorker was a member of ScriptRunner but was only used in its constructor so I made it a local variable.

Reply:
The overall understanding of the code is probably better if isWorker is defined with the other two, even if it is not used (yet?).